### PR TITLE
Don't override a dirty flag from true to false

### PR DESCRIPTION
**Pull request by arajasek**
Fix from @Kubuxu

### CODE REVIEW

File: hamt.go [hunk 1]
```diff
@@ -646,8 +646,10 @@ func (n *Node) modifyValue(ctx context.Context, hv *hashBits, k []byte, v *cbg.D
 			return UNMODIFIED, err
 		}
 
-		child.dirty = bool(modified)
-
+		if !child.dirty {
+			child.dirty = bool(modified)
+		}
```
Comment (Kubuxu) [inline hamt.go:651]: `child.dirty |= bool(modified)`

Comment (Kubuxu) [inline hamt.go:651]: or `child.dirty = child.ditrty || bool(modified)`

**Comment by Kubuxu** [comment]
Can someone try creating a test for the dirty flag being overwritten.

**Comment by Kubuxu** [comment]
Most likely entry point is `SetIfAbsent`.

File: hamt.go (hunk 1 above)
Comment (Stebalien) [reply hamt.go:650]: Or, even better:

```go
if modified { child.dirty = true }
```

**Review by Stebalien — APPROVED** [review]

**Pull request merged**